Source/ARTPresenceMap.m

```diff
@@ -150,9 +150,22 @@ - (void)leaveMembersNotPresentInSync {
     }
 }
 
+- (void)reset {
+    _members = [NSMutableDictionary dictionary];
+    _localMembers = [NSMutableDictionary dictionary];
+}
+
 - (void)reenterLocalMembersMissingFromSync {
     [_logger debug:__FILE__ line:__LINE__ message:@"%p reentering local members missed from sync (syncSessionId=%lu)", self, (unsigned long)_syncSessionId];
-    NSSet *filteredLocalMembers = [_localMembers filteredSetUsingPredicate:[NSPredicate predicateWithFormat:@"syncSessionId != %lu", (unsigned long)_syncSessionId]];
+    NSMutableDictionary<NSString *,ARTPresenceMessage *> *filteredLocalMembers = [NSMutableDictionary dictionary];
+    
+    [_localMembers enumerateKeysAndObjectsUsingBlock:^(NSString *key, ARTPresenceMessage *message, BOOL* stop) {
+        if ((unsigned long)_syncSessionId != message.syncSessionId) {
+            filteredLocalMembers[key] = message;
+        }
```

Comment:
What is this syncSessionId mechanism doing, what's its purpose?

Reply:
If it's to only "reenterLocalMembersMissingFromSync", then it shouldn't be doing that that -- RTP17g requires the lib to do a re-enter `for each member of the RTP17 internal PresenceMap`, no mention about only members that were missing from the sync. That was the main change between RTP17d and RTP17g

Reply:
This syncSessionId is a number that is initialized at 0 and increments by one on each `startSync` call. So yes, it seems to be doing what you defined above. I fixed this dad3c14d0918d10faa24bd17598399ce4c6a08a2 but also removed dependence of reentry from `endSync` and it happens after a potential `endSync` call happens. Is there a certain order of actions that needs to be followed or is putting this after a potential `endSync` ok?